tests/tests.cpp

```diff
@@ -166,7 +166,7 @@ TEST(system_lib, basic_test) {
 #endif
 }
 
-#if ((defined(_MSVC_LANG) && _MSVC_LANG >= 201703L) || __cplusplus >= 201703L)
+#if (defined(DYLIB_CPP17))
```

Comment:
`DYLIB_CPP17` is undef at the end of `dylib.hpp` so this condition will always be false.
You should undo this change on the test source file